LanguageFeatures/Patterns/scope_A02_t01.dart

```diff
@@ -18,12 +18,15 @@
 
 // SharedOptions=--enable-experiment=patterns
 
+import "../../Utils/expect.dart";
+
 const c = 1;
 
 main() {
-  for (var [c] = [c] in [[]]) {
-//                ^
-// [analyzer] unspecified
-// [cfe] unspecified
+  bool visited = false;
+  for (var [c] in [[]]) {
```

Comment:
I think the only entity that could play the role as the "initializer" for the iteration variable of a for-in statement is the iterable. 

However, the specification of a for-in statement uses a desugaring where there is no name clash, because the iterable is evaluated outside the while statement of the desugared code, and the iteration variable is declared inside that while statement.

In other words, we should probably have

  const c = [2];

  void main() {
    bool visited = false;
    for (var [c] in [c]) {
      Expect.listEquals([2], c);
      visited = true;
    }
    Expect.isTrue(visited);
  }

It looks funny, but the desugaring would actually put the iterable in the scope that makes `c` refer to the top-level variable (or whatever `c` resolves to when it is outside the loop itself).

I wonder if we can find anyone at all who doesn't think that's weird. ;-)

Reply:
Oh, and the description should be adjusted to say that there are no errors here.